### ISSUE TITLE
Symfony 3.4: Remote_route is not being used

### ISSUE THREAD

**Issue by Horzta** [tetranz/select2entity-bundle#130]
I installed this and tried to search but the ajax call is using the same url as the current page, not the one I mentioned at the remote_route:

Supposed to be: `http://127.0.0.1:8000/asset/api/asset_list/?field_name=assets`
But it shows:  `http://127.0.0.1:8000/user/19?field_name=assets`

The current page that the ajax is being called from is `user/19`

Here is my code, please help:

```
 /**
     * Lists all asset entities.
     *
     * @Route("/api/asset_list", name="api_asset_list")
     * @Method("GET")
     */
    public function jsonAction()
    {
        $em = $this->getDoctrine()->getManager();
        $assets = $em->getRepository('AppBundle:Asset')->findAll();

        $assetjson = array();
        foreach($assets as $asset)
            $assetsjson[] = array('id' => $asset->getId(), 'text'=> $asset->getCode() . " - ". $asset->getDescription() . " - ". $asset->getLocation());

        $response = new Response();
        $response->setContent(json_encode($assetsjson));

        $response->headers->set('Content-Type', 'application/json');
        return $response;
    }
```

```
        $builder->add('assets', Select2EntityType::class, [
                    'multiple' => true,
                    'remote_route' => 'api_asset_list',
                    'class' => 'AppBundle\Entity\Asset',
                    'minimum_input_length' => 10,
                    'page_limit' => 10,
                    'allow_clear' => true,
                    'delay' => 250,
                    'language' => 'en',
                    'placeholder' => 'Select assigned assets',
                    // 'object_manager' => $objectManager, // inject a custom object / entity manager 
                ]);
```

```
    <div class = "container">
        <div class = "row title">
            <h5 class = "modal-title">Assign Assets</h5>
        </div>
        <div class = "row">
            {{form_start(assign_form)}}
            {{form_widget(assign_form)}}
            <input class = "btn btn-block btn-success" type="submit" value="Assign" />
            {{form_end(assign_form)}}
        </div>
    </div> 
{% endblock %}
{% block javascripts %}
    <script src="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.6-rc.0/js/select2.min.js"></script>
    <script src="{{ asset('bundles/tetranzselect2entity/js/select2entity.js') }}"></script>
{% endblock %}
```

```
# Twig Configuration
twig:
    debug: '%kernel.debug%'
    strict_variables: '%kernel.debug%'
    form_themes: ['bootstrap_3_layout.html.twig','TetranzSelect2EntityBundle:Form:fields.html.twig']
    cache:  false
```

It also seems to be ignoring the minimum input length.

**Comment by Horzta** [comment]
So after some fiddling I think there is something with select2 4.0.6 that it doesn't use the variables at

`<select id="appbundle_assetassign_assets" name="appbundle_assetassign[assets][]" data-ajax--url="/asset/api/asset_list?page_limit=10" data-ajax--cache="false" data-ajax--cache-timeout="0" data-ajax--delay="250" data-ajax--data-type="json" data-language="en" data-minimum-input-length="10" data-placeholder="Select assigned assets" data-page-limit="10" data-scroll="true" data-autostart="true" class="select2entity form-control select2-hidden-accessible" data-name="assets" multiple="" data-allow-clear="true" tabindex="-1" aria-hidden="true"><option value=""></option><option value="134" selected="selected">CE-PS-0047</option><option value="237" selected="selected">IF-MSE-0052</option><option value="325" selected="selected">IF-MON-0022</option><option value="390" selected="selected">FF-LCR-A-03</option><option value="546" selected="selected">IF-LT-0016</option><option value="563" selected="selected">IF-KB-0017</option><option value="789" selected="selected">IF-CPU-0018</option><option value="921" selected="selected">FF-OC-0106</option><option value="1059" selected="selected">IF-AVR-0008</option></select>`

In my case the `data-ajax--url` variable.

I switched to 4.0.5 and it worked a charm.

**Comment by Reileb** [comment]
This issue is not solved, i too have it with 4.0.6-rc.1. The used ajax URL is the one of the current page.
